File: win-prog/ProgWinCS/Lines Curves and Area Fills/DashedEllipse/DashedEllipse.cs
//--------------------------------------------
// DashedEllipse.cs © 2001 by Charles Petzold
//--------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class DashedEllipse: PrintableForm
{
     public new static void Main()
     {
          Application.Run(new DashedEllipse());
     }
     public DashedEllipse()
     {
          Text = "Dashed Ellipse Using DrawArc";
     }
     protected override void DoPage(Graphics grfx, Color clr, int cx, int cy)
     {
          Pen       pen  = new Pen(clr);
          Rectangle rect = new Rectangle(0, 0, cx - 1, cy - 1);

          for (int iAngle = 0; iAngle < 360; iAngle += 15)
               grfx.DrawArc(pen, rect, iAngle, 10);
     }
}

File: win-prog/ProgWinVB/Lines Curves and Area Fills/DashedEllipse/DashedEllipse.vb
'----------------------------------------------
' DashedEllipse.vb (c) 2002 by Charles Petzold
'----------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class DashedEllipse
    Inherits PrintableForm

    Shared Shadows Sub Main()
        Application.Run(New DashedEllipse())
    End Sub

    Sub New()
        Text = "Dashed Ellipse Using DrawArc"
    End Sub

    Protected Overrides Sub DoPage(ByVal grfx As Graphics, _
            ByVal clr As Color, ByVal cx As Integer, ByVal cy As Integer)
        Dim pn As New Pen(clr)
        Dim rect As New Rectangle(0, 0, cx - 1, cy - 1)
        Dim iAngle As Integer

        For iAngle = 0 To 345 Step 15
            grfx.DrawArc(pn, rect, iAngle, 10)
        Next iAngle
    End Sub
End Class
